src/transaction_builder.cpp

```diff
@@ -536,11 +549,11 @@ TransactionBuilderResult TransactionBuilder::Build()
         } else if (firstOrchardSpendAddr.has_value()) {
             auto ovk = orchardSpendingKeys[0].ToFullViewingKey().ToInternalOutgoingViewingKey();
             AddOrchardOutput(ovk, firstOrchardSpendAddr.value(), change, std::nullopt);
-        } else if (!spends.empty()) {
-            auto fvk = spends[0].expsk.full_viewing_key();
-            auto note = spends[0].note;
-            libzcash::SaplingPaymentAddress changeAddr(note.d, note.pk_d);
-            AddSaplingOutput(fvk.ovk, changeAddr, change, std::nullopt);
+        } else if (firstSaplingSpendAddr.has_value()) {
```

Comment:
I'm surprised that this code is still in here; it should probably be removed, with change address selection done by `WalletTxBuilder`. I wonder if this code is even ever invoked any more?

Reply:
Probably used in tests. But in any case, there is likely still some dead code that can be removed; I just didn't have an easy way (that I knew of) to identify it other than by eye (unlike with Rust where `cargo` warns you).